codice/ddf
# DDF-4038 Improving sorting of Gazetteer suggestions

#### What does this PR do?
When doing a keyword search currently the gazetteer data isn't very intuitive. I.E. Paris, Texas comes up before Paris, France, Chinatown, USA comes up before China, etc. This PR sorts suggestions as follows, countries first then geonames data is sorted based on population. Hopefully, this will result in the suggester being a little more intuitive. Currently my implementation creates a score on ingest giving a score of 11 to any country and then creating a score from 2 - 10 based on population, then sorts the suggestions based on their perspective score in descending order. 
#### Who is reviewing it? 
@adimka @GabrielFabian 

#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining @millerw8 

#### How should this be tested?
Ingest Gazetteer data 
 go to search ui and do a basic search do location (Somewhere specific) and then do a keyword search
Type in two letters and see the suggestions and ensure that countries come first 
(i.e) when typing in `Ar` you should see Argentina, Armenia, followed by some cities 

#### Any background context you want to provide?
#### What are the relevant tickets?
[DDF-4038](https://codice.atlassian.net/browse/DDF-4038)
#### Screenshots

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
